include/DoocsBackendNumericRegisterAccessor.h

```diff
@@ -0,0 +1,371 @@
+        memcpy(this->buffer_2D[0].data(), sourcePointer, this->nElements * sizeof(SourceType));
+      }
+      else {
+        std::transform(sourcePointer, sourcePointer + this->nElements, this->buffer_2D[0].begin(),
+            [](const SourceType& v) { return ChimeraTK::numericToUserType<UserType>(v); });
+      }
+    };
+
+    // optimise depending on type
+    switch(this->dst.type()) {
+      case DATA_SPECTRUM:
+      case DATA_A_FLOAT: {
+        copyFromSourcePointer(this->dst.get_float_array());
+        return;
+      }
+      case DATA_A_DOUBLE: {
+        copyFromSourcePointer(this->dst.get_double_array());
+        return;
+      }
+      case DATA_A_INT: {
+        copyFromSourcePointer(this->dst.get_int_array());
+        return;
+      }
+      case DATA_A_LONG: {
+        copyFromSourcePointer(this->dst.get_long_array());
+        return;
+      }
+      case DATA_A_BOOL:
+      case DATA_A_SHORT: {
+        copyFromSourcePointer(this->dst.get_short_array());
+        return;
+      }
+      case DATA_IFFF: {
+        for(size_t i = 0; i < this->nElements; i++) {
+          if(i + this->elementOffset == 0) {
+            this->buffer_2D[0][i] = ChimeraTK::numericToUserType<UserType>(dataGet<int>(i + this->elementOffset));
+          }
+          else {
+            this->buffer_2D[0][i] = ChimeraTK::numericToUserType<UserType>(dataGet<float>(i + this->elementOffset));
+          }
+        }
+        return;
+      }
+      default:
+        // inefficient copying via single element access for other data types (including scalars)
+        callForDoocsType(this->dst, [&](auto t) {
+          using T = decltype(t);
+          for(size_t i = 0; i < this->nElements; i++) {
+            this->buffer_2D[0][i] = ChimeraTK::numericToUserType<UserType>(dataGet<T>(i + this->elementOffset));
+          }
+        });
+    }
+  }
+
+  /********************************************************************************************************************/
+
+  template<>
+  inline void DoocsBackendNumericRegisterAccessor<ChimeraTK::Void>::doPreWrite(
+      TransferType type, VersionNumber version) {
+    DoocsBackendRegisterAccessor<ChimeraTK::Void>::doPreWrite(type, version);
+
+    src.set(0);
+  }
+
+  /********************************************************************************************************************/
+
+  template<typename UserType>
+  void DoocsBackendNumericRegisterAccessor<UserType>::doPreWrite(TransferType type, VersionNumber version) {
+    DoocsBackendRegisterAccessor<UserType>::doPreWrite(type, version);
+
+    // implement read-modify-write
+    if(this->isPartial) {
+      this->doReadTransferSynchronously();
+      this->src = this->dst;
+      // make sure the array is long enough, as the remote server might have changed the length on its side
+      if(size_t(this->src.length()) < this->nElements + this->elementOffset) {
+        this->src.length(this->nElements + this->elementOffset);
```

Comment:
Can you simply send something longer than what the server had before?

Reply:
Yes, you can even send completely different data types. It is up to the server to check this and return an error if it cannot handle this.